tests/lib/rules/no-side-effects-wait-for.test.ts

```diff
@@ -0,0 +1,210 @@
+      `,
+    },
+    {
+      code: `
+        import { waitFor } from '@testing-library/react';  
+        await waitFor(() => {})
+      `,
+    },
+    {
+      code: `
+        import { waitFor } from '@testing-library/react';  
+        await waitFor(function() {})
+      `,
+    },
+    {
+      code: `
+        import { waitFor } from '@testing-library/react';  
+        await waitFor(() => {
+          // testing
+        })
+      `,
+    },
+    {
+      code: `
+        import { waitFor } from '@testing-library/react';  
+        await waitFor(function() {
+          // testing
+        })
+      `,
+    },
+    {
+      code: `
+        import { waitFor } from '@testing-library/react';  
+        fireEvent.keyDown(input, {key: 'ArrowDown'})
+        await waitFor(() => {
+          expect(b).toEqual('b')
+        })
+      `
+    }, {
+      code: `
+        import { waitFor } from '@testing-library/react';  
+        fireEvent.keyDown(input, {key: 'ArrowDown'})
+        await waitFor(function() {
+          expect(b).toEqual('b')
+        })
+      `
+    }
+  ],
+  invalid: [
+    // fireEvent
+    {
+      code: `
+        import { waitFor } from '@testing-library/react';  
+        await waitFor(() => {
+          fireEvent.keyDown(input, {key: 'ArrowDown'})
+        })
+      `,
+      errors: [{ messageId: 'noSideEffectsWaitFor' }]
+    },
+    {
+      code: `
+        import { waitFor } from '@testing-library/react';  
+        await waitFor(() => {
+          expect(b).toEqual('b')
+          fireEvent.keyDown(input, {key: 'ArrowDown'})
+        })
+      `,
+      errors: [{ messageId: 'noSideEffectsWaitFor' }]
+    },
+    {
+      code: `
+        import { waitFor } from '@testing-library/react';  
+        await waitFor(() => {
+          fireEvent.keyDown(input, {key: 'ArrowDown'})
+          expect(b).toEqual('b')
+        })
+      `,
+      errors: [{ messageId: 'noSideEffectsWaitFor' }]
+    },
+    {
+      code: `
+        import { waitFor } from '@testing-library/react';  
+        await waitFor(function() {
+          fireEvent.keyDown(input, {key: 'ArrowDown'})
+        })
+      `,
+      errors: [{ messageId: 'noSideEffectsWaitFor' }]
+    },
+    {
+      code: `
+        import { waitFor } from '@testing-library/react';  
+        await waitFor(function() {
+          expect(b).toEqual('b')
+          fireEvent.keyDown(input, {key: 'ArrowDown'})
+        })
+      `,
+      errors: [{ messageId: 'noSideEffectsWaitFor' }]
+    },
+    {
+      code: `
+        import { waitFor } from '@testing-library/react';  
+        await waitFor(function() {
+          fireEvent.keyDown(input, {key: 'ArrowDown'})
+          expect(b).toEqual('b')
+        })
+      `,
+      errors: [{ messageId: 'noSideEffectsWaitFor' }]
+    },
+    // userEvent
+    {
+      code: `
+        import { waitFor } from '@testing-library/react';  
+        await waitFor(() => {
+          userEvent.keyDown(input, {key: 'ArrowDown'})
```

Comment:
same in these comments - just to make them more realistic, let's better use the methods from userEvent API